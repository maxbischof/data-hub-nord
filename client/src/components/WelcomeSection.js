import React from 'react'
import styled from 'styled-components'
import welcomeImage from '../images/sea.jpg'
import Button from './ui/Button'
import PropTypes from 'prop-types'

export default function WelcomeSection({ onClickButton, scrollTo }) {
  return (
    <StyledWelcomeSection>
      <Headline>Entdecke Open Data in Norddeutschland</Headline>
      <SubHeadline>
        Visualisierung und Zugang zu den wichtigsten Daten
      </SubHeadline>
      <Button
        styleType="action"
        onClick={onClickButton}
        onClickParameter={scrollTo}
      >
        Daten anzeigen
      </Button>
    </StyledWelcomeSection>
  )
}

WelcomeSection.propTypes = {
  onClickButton: PropTypes.func,
  scrollTo: PropTypes.object,
}

const StyledWelcomeSection = styled.section`
  height: 500px;
  background: url("${welcomeImage}");
  background-position: bottom;
  background-size: cover;
  display: flex;
  flex-direction: column;
  align-items: center;
  flex-shrink: 0;
`

const Headline = styled.h1`
  color: white;
  font-size: 35px;
  font-weight: 100;
  text-align: center;
  padding: 0 10px;
  margin: 120px 0 10px 0;
`

const SubHeadline = styled.h2`
  color: white;
  font-size: 20px;
  text-align: center;
  font-weight: 400;
  margin: 0 0 50px 0;
  padding: 0 10px;
`
