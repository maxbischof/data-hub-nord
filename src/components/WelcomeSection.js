import React from 'react'
import styled from 'styled-components'
import welcomeImage from "../images/sea.jpg"
import WelcomeButton from "./ui/WelcomeButton"
import PropTypes from 'prop-types'

export default function WelcomeSection({ onClickButton, scrollTo }) {
  return (
    <StyledWelcomeSection>
      <Headline>Entdecke Open Data in Norddeutschland</Headline>
      <SubHeadline>
        Visualisierung und Zugang zu den wichtigsten Daten
      </SubHeadline>
      <WelcomeButton color="#e63946" onClick={onClickButton} reference={scrollTo}>Daten anzeigen</WelcomeButton>
    </StyledWelcomeSection>
  )
}

WelcomeSection.propTypes = {
  onClickButton: PropTypes.func,
  scrollTo: PropTypes.object
}

const StyledWelcomeSection = styled.section`
  height: 500px;
  background: url("${welcomeImage}");
  background-position: bottom;
  background-size: cover;
  display: flex;
  flex-direction: column;
  align-items: center;
  flex-shrink: 0;
`

const Headline = styled.h1`
  color: white;
  font-size: 35px;
  font-weight: 100;
  text-align: center;
  padding: 0 10px 0 10px;
  margin: 120px 0 10px 0;
`

const SubHeadline = styled.h2`
  color: white;
  font-size: 20px;
  text-align: center;
  font-weight: 400;
  margin: 0 0 50px 0;
  padding: 0 10px 0 10px;
`