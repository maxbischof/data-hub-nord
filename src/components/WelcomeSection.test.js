import React from "react"
import WelcomeSection from "./WelcomeSection"
import renderer from "react-test-renderer"

it("renders correctly", () => {
  const datasetListRef = null

  const scrollToRef = (ref) => {
    console.log("scroll")
    window.scrollTo(0, ref.current.offsetTop)
  }

  const tree = renderer
    .create(<WelcomeSection onClickButton={scrollToRef} scrollTo={datasetListRef}/>)
    .toJSON()
  expect(tree).toMatchSnapshot()
})
