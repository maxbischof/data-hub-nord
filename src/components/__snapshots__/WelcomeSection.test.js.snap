// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<section
  className="sc-AxirZ dwIUgl"
>
  <h1
    className="sc-AxiKw gshFvr"
  >
    Entdecke Open Data in Norddeutschland
  </h1>
  <h2
    className="sc-AxhCb iupUPJ"
  >
    Visualisierung und Zugang zu den wichtigsten Daten
  </h2>
  <button
    className="sc-AxjAm faIUmi"
    color="#e63946"
    onClick={[Function]}
  >
    Daten anzeigen
  </button>
</section>
`;
